### DISCUSSION TITLE
Interesting Observation: Rounded versus Square Corners on Calendar

### DISCUSSION THREAD

**Discussion by DavidBerdik** [valinet/ExplorerPatcher#439] [General]
Hello all,

I have noticed a rather interesting behavior with the user interface for the calendar that I would like to ask about. I would not go so far as to call it a bug, but an interesting quirk that I would be curious to be given a technical explanation for. The quirk has to do with the user interface for the calendar.

On Windows 10, the numbers on the calendar flyout are surrounded with a square border, like the following:

![image](https://user-images.githubusercontent.com/7663225/142742560-6ce55094-8022-4053-a689-c12fc43ae074.png)

On Windows 11, the numbers on the calendar flyout are surrounded with a circular border, like the following:

![image](https://user-images.githubusercontent.com/7663225/142742567-c156094e-55ca-4ca7-80db-765bc7e3df66.png)

The interesting quirk that I have noticed is that when ExplorerPatcher is installed, the square versus rounded styling changes seemingly arbitrarily each time Explorer is launched. Additionally, the styling that is applied is consistent between the Windows 10 and Windows 11 calendar UIs. For example, here is a screenshot of the Windows 10 and Windows 11 calendar flyouts from my Windows 11 VM with ExplorerPatcher installed.

![image](https://user-images.githubusercontent.com/7663225/142742620-5e84c4be-3186-4c9b-b057-ed08790600a0.png)

![image](https://user-images.githubusercontent.com/7663225/142742629-06cd21ad-e2c8-4af4-9aeb-adbdb3e9a1ee.png)

In this case, the Windows 10-style square corners have been applied to both calendars.

Let us now look at the way the calendar flyouts appear on my PC running ExplorerPatcher on Windows 11.

![image](https://user-images.githubusercontent.com/7663225/142742741-9c0f25ec-f85d-4100-af3c-33e0272f5263.png)

![image](https://user-images.githubusercontent.com/7663225/142742747-07315eef-f02f-47bb-b719-831406b61ced.png)

In this case, the Windows 11-style rounded corners were used, but I have seen the square corners used sometimes after reboots as well. It seems to be random.

As I have said, I don't really care about the difference between the corners, nor do I consider it a bug, but I would be curious as to the reason behind why this could be happening if anyone has any ideas.

**Comment by valinet** [comment]
Yeah, I am curious too, I think some guy at Microsoft, preferably someone that worked on these, is in a better position to answer rather any of us here or me, who I am simply learning as I go when trying to do various stuff. Personally, I don’t know, it’s not my implementation, and it’s pretty hard to do the detective work and figure it out from the outside either.

It’s an interesting observation, but yeah, idk the answer. I am also more curious why it randomly picks square or rounded UI and whether this ever happens on the stock (Windows 11 without EP) UI as well. Or in SAB. Anyway, yeah, if someone would take the time to investigate, some answers would surface, without, we’re all left wondering.

Thanks

**Reply by DavidBerdik** [reply]
That's quite understandable. I honestly would have been surprised if you were to have an answer, but thought I'd ask anyway! 🙂

One other interesting thing that I have noticed from playing around is that it looks like the square corner is more likely to occur. From playing around in my virtual machine as well as rebooting my PC several times, it seems that while the rounded corners do occur sometimes, they are much less often.
